github-adapter/src/main/java/be/xplore/githubmetrics/githubadapter/GithubApiUtilities.java

```diff
@@ -2,69 +2,27 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
-import org.springframework.web.client.RestClient;
-import org.springframework.web.util.UriBuilder;
 
-import java.net.URI;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
 import java.util.function.Function;
 
 @Component
 public class GithubApiUtilities {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GithubApiUtilities.class);
-    private final RestClient restClient;
-
-    public GithubApiUtilities(
-            @Qualifier("defaultRestClient") RestClient restClient
-    ) {
-        this.restClient = restClient;
-    }
-
-    public Function<UriBuilder, URI> setPathAndParameters(
-            String path,
-            List<Object> uriVariables
-    ) {
-        return uriBuilder ->
-                uriBuilder.path(path)
-                        .build(uriVariables.toArray(new Object[0]));
-    }
-
-    public Function<UriBuilder, URI> setPathAndParameters(
-            String path,
-            List<Object> uriVariables,
-            Map<String, String> parameters
-    ) {
-        return uriBuilder -> {
-            uriBuilder.path(path);
-            parameters.forEach(uriBuilder::queryParam);
-            return uriBuilder.build(uriVariables.toArray(new Object[0]));
-        };
-    }
-
-    public Function<UriBuilder, URI> setPathAndParameters(
-            String path,
-            Map<String, String> parameters
-    ) {
-        return uriBuilder -> {
-            uriBuilder.path(path);
-            parameters.forEach(uriBuilder::queryParam);
-            return uriBuilder.build();
-        };
-    }
 
+    @SuppressWarnings("checkstyle:ParameterNumber")
     public <D, A> List<D> followPaginationLink(
```

Comment:
in the end didn't create an entry function? Like one where you don't have to enter the page number?

Reply:
no. could be a potential optimization but with time allowed I left it as is.